homeassistant/helpers/entity.py

```diff
@@ -971,7 +972,12 @@ def async_write_ha_state(self) -> None:
         """Write the state to the state machine."""
         if self.hass is None:
             raise RuntimeError(f"Attribute hass is None for {self}")
-
+        if (
```

Comment:
In the future, we could add a fast check to entity sources to see if it's a custom component and always do this check if it is

Reply:
I very much like that idea.

Reply:
And it will be easy to search the code of custom components for calling anything `_internal`.

Reply:
Started working on that in https://github.com/home-assistant/core/pull/116027

Reply:
https://github.com/home-assistant/core/pull/116044

Reply:
https://github.com/home-assistant/core/pull/116044/commits/ab0c620c22b3132de45e9e68fd469ae4f4dfe84a will implement that